python/cudf/cudf/core/mixins/binops.py

```diff
@@ -59,7 +59,17 @@ def _binaryop(self, other, op: str):
     Must be overridden by subclasses, the default implementation raises a
     NotImplementedError.
     """
-    raise NotImplementedError
+    if op == "__eq__":
+        raise TypeError(
+            "'==' not supported between instances of "
+            f"'{type(self).__name__}' and '{type(other).__name__}'"
+        )
+    if op == "__ne__":
+        raise TypeError(
+            "'!=' not supported between instances of "
+            f"'{type(self).__name__}' and '{type(other).__name__}'"
+        )
+    raise NotImplementedError()
```

Comment:
Would it be cleaner to leave `BinaryOperand._binaryop` as-is, and just patch the `__eq__` and `__ne__`  methods? Or do we need to change all three methods?